WooCommerce: Add the order list view

ryelle opened the pull request:
Fixes #13654 

<img width="1180" alt="screen shot 2017-06-20 at 9 47 49 am" src="https://user-images.githubusercontent.com/541093/27336461-a1d8291c-559d-11e7-94c0-5abb4f7f5c32.png">

This adds the basic orders list view. We've decided pagination will be part of v2, so right now you can only see a max of 100 orders. This particular PR just gets the basic display in, I'll add follow-ups for better currency display, and for the split status bar (in https://github.com/Automattic/wp-calypso/issues/13654#issuecomment-306613311).

To test:

- Visit `https://calypso.live/store/orders/$site`
- You should see the list of orders on your site

a8ci18n commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,180 @@
+		const headers = (
+			<TableRow>
+				<TableItem isHeader>
+					<FormInputCheckbox aria-label={ translate( 'Select All', {
```
Hi! I've found a possible matching string that has already been translated **95** times:
`translate( 'Select all' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This string already exists without a **context**. Only add a context if the meaning of the string is very specific.

a8ci18n commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,180 @@
+		return (
+			<TableRow key={ i }>
+				<TableItem className="orders__table-checkbox">
+					<FormInputCheckbox aria-label={ translate( 'Select order %(order)s', {
```
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

a8ci18n commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,180 @@
+			case 'cancelled':
+				return <span className={ classes }>{ translate( 'Cancelled' ) }</span>;
+			case 'refunded':
+				return <span className={ classes }>{ translate( 'Refunded' ) }</span>;
```
Hi! I've found a possible matching string that has already been translated **37** times:
`translate( 'Refunded To' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,180 @@
+			case 'on-hold':
+				return <span className={ classes }>{ translate( 'On hold' ) }</span>;
+			case 'completed':
+				return <span className={ classes }>{ translate( 'Completed' ) }</span>;
```
Hi! I've found a possible matching string that has already been translated **52** times:
`translate( '&laquo; completed' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

kellychoffman commented:
Awesome!!! Going to style this up, but one comment before I do: 

Since we aren't including any bulk editing for v1, let's get rid of the column of checkboxes.

kellychoffman commented:
When there are no orders, can we use `EmptyContent` [#](https://wpcalypso.wordpress.com/devdocs/design/empty-content) like we do in the product list view? (Apologize for not putting this in the design)

Copy:

```Orders will appear here as they come in.

[Manually add an order]```

ryelle commented:
I've added the EmptyContent component, and also split out the the status labels — I'm not sure I got the labels right, so I could use a second look (from anyone) at how they're generated (https://github.com/Automattic/wp-calypso/pull/14871/commits/c00ba35d3fdbbe9ea4d845d7e98ca2e7b90b91e4).

a8ci18n commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,194 @@
+		let shippingLabel;
+		switch ( status ) {
+			case 'pending':
+				shippingLabel = translate( 'New order' );
```
Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Save New Order' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

kellychoffman commented:
Split labels are looking good! 

Could you remove the sectionNav above emptyContent?

allendav commented:
I was surprised to see two states for this order on octopus - was expecting just "on-hold" for it

<img width="667" alt="screen shot 2017-06-20 at 3 58 12 pm" src="https://user-images.githubusercontent.com/1595739/27359583-6533a8b8-55d1-11e7-8615-1ec4459d5526.png">

allendav commented:
@kellychoffman / @jameskoster - would the order date be a useful place for moment instead? (e.g. 1 hour ago, 1 day ago, etc etc)

<img width="667" alt="screen shot 2017-06-20 at 4 02 35 pm" src="https://user-images.githubusercontent.com/1595739/27359689-1aa81968-55d2-11e7-88e4-676d7acc343d.png">

allendav commented on client/extensions/woocommerce/app/orders/index.js:
```diff
@@ -0,0 +1,36 @@
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
```
Would the fallback selector be more robust here?  https://github.com/Automattic/wp-calypso/blob/e1f9cddd82b2ca92e1e3f20f7cbec36812ea088b/client/extensions/woocommerce/state/sites/selectors.js#L16

allendav commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,205 @@
+				shippingLabel = translate( 'New order' );
+				paymentLabel = translate( 'Paid in full' );
+				break;
+			case 'on-hold':
```
Ah, now I see why we were getting two labels for some orders. Nevermind my other comment :P

allendav commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,205 @@
+
+export default connect(
+	state => {
+		const siteId = getSelectedSiteId( state );
```
See fallback suggestion above

allendav commented:
The order state needs a little love on narrowish screens:

<img width="616" alt="screen shot 2017-06-20 at 4 15 12 pm" src="https://user-images.githubusercontent.com/1595739/27360105-2aa929ea-55d4-11e7-80b9-4188fa42bfd6.png">

cc @kellychoffman @jameskoster

allendav commented:
If I:

- go to http://calypso.localhost:3000/store/orders/{siteslug}
- indexedDB.deleteDatabase( 'calypso' );
- localStorage.clear()
- refresh (Command-Shift-R)

The body fails to render after the sidebar eventually does and I get:

<img width="1223" alt="screen shot 2017-06-20 at 4 21 35 pm" src="https://user-images.githubusercontent.com/1595739/27360232-c0434dc8-55d4-11e7-9e7d-3d792ab069d1.png">

kellychoffman commented:
> The order state needs a little love on narrowish screens:

I tried a bunch of things, but can't figure out how to have a max-width on the table so that it horizontally scrolls on screens smaller than 960px. We have this on the variation table in add-product, but I can't get it to apply here. I think its because the variations table is within a `Card`?

jameskoster commented:
@ryelle could we make it so that the entire row links to the order rather than just the number?

ryelle commented:
> @ryelle could we make it so that the entire row links to the order rather than just the number?

Probably, I can look into that when I do the single-view PR

ryelle commented:
I've switch to the fallback selector and fixed the undefined site issues, but I think the other concerns can be handled in another PR — would love to merge this before I need to rebase it again 😄

kellychoffman commented on client/extensions/woocommerce/app/orders/orders-list.js:
```diff
@@ -0,0 +1,178 @@
+				<TableItem className="orders__table-status">
+					{ this.getOrderStatus( order.status ) }
+				</TableItem>
+				<TableItem className="orders__table-total">
```
I faked this for now, but would be good to get in another PR.

kellychoffman commented:
> @ryelle could we make it so that the entire row links to the order rather than just the number?

This is also done in Products list. These tables should be as consistent as possible, but fine to do in another PR.

kellychoffman commented:
> I tried a bunch of things, but can't figure out how to have a max-width on the table so that it horizontally scrolls on screens smaller than 960px. We have this on the variation table in add-product, but I can't get it to apply here. I think its because the variations table is within a Card?

Followup on this: Jay addressed this in https://github.com/Automattic/wp-calypso/pull/15351

ryelle commented:
>This is also done in Products list. These tables should be as consistent as possible, but fine to do in another PR.

I just want to double check the accessibility of linking an entire table row, and possibly I'll update the shared table component to make this super-easy in all our tables. But that's why I said another PR 🙂
